### ISSUE TITLE
[Bug]: IsFocusVisible is not shared between component surfaces

### ISSUE THREAD

**Issue by micahgodbolt** [microsoft/fluentui#25104]
### Library

React  / v8 

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/micahgodbolt/pen/YzLjdoY

## Actual Behavior
When I keyboard to focus on a button, the button gets a focus rectangle due to isFocusVisible class on the ThemeProvider.

When clicking this button opens up a panel (or modal or anything else on another layer), the isFocusVisible className is not applied to that surface, and therefor the focus rectangles do not show up.

## Expected Behavior
Focus rectangles should show up when focus moves between button and panel (or modal etc)

### Requested priority

Normal

**Comment by micahgodbolt** [comment]
Options:
1. copy isFocusVisible into the ThemeProvider context so that other surfaces can look for that value in context
2. duplicate styles with focus-visible so that modern browsers always get the correct styles

**Comment by micahgodbolt** [comment]
For instance, when you open a dialog from a panel, the panel has focus visible, but the opened dialog doesn't know that focus should be visible

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/1434956/194177162-2e2e57e8-b819-4a9b-bfc0-2e6d0f3cd4f8.png)

**Comment by behowell** [comment]
PR https://github.com/microsoft/fluentui/pull/24340 made it so updates to one layer's isFocusVisible will also update any child's isFocusVisible. However, it looks like it's possibly not getting the initial state correct when the layer is created. The fix here might be to make sure that the child layer gets focus styling when it is created?

**Comment by swese44** [comment]
This ticket should be tagged as an a11y issue, we're catching a11y tickets in Yammer/Viva Engage. @micahgodbolt and @behowell how can we get this bug prioritized?

For more background the `ms-Fabric--isFocusVisible` className used to get applied to the body element. After some digging it looks like this behavior was changed in #24025. We have other ongoing conversations about the FUI8 LayerHost and FUI9 PortalHost divs getting rendered as a child of the body element instead of a child of our app root, this is a) causing FUI9 components to not be themed when rendered in FUI8 layers and b) prevents Chromatic (via React Testing Library) to be unable to access DOM outside of the app root which prevents us from capturing visual diff regression tests for any Layer/Portal content.

It seems that all of these issues would be resolved by keeping the LayerHost and PortalHost divs in the app root. Can Fluent provide FUI8 LayerElementProvider and FUI9 PortalElementProvider components so we can "globally" set all Layers/Portals to be rendered within our app root?

**Comment by smhigley** [comment]
Fixed by #28157
